### ISSUE TITLE
Date format looks incorrect in report list's title.

### ISSUE THREAD

**Issue by skrishn** [Esri/solutions-components#611]
### Describe the bug

The date format in the report list looks incorrect. 

### Expected Behavior

It should appear in the same format as in the webmap popup. 

### Reproduction Steps & Sample

https://localgovdev.mapsdevext.arcgis.com/apps/instant/reporter/index.html?appid=f2897862c5574d4bb4b607a105884bf9

Webmap popup

![image](https://github.com/Esri/solutions-components/assets/3836616/f04846db-6fe3-4412-8031-9289a85fc17c)

Report list date format 

![image](https://github.com/Esri/solutions-components/assets/3836616/cd5dc22a-dd43-4809-b4a7-f8df932ca1d8)

Popup title date format

![image](https://github.com/Esri/solutions-components/assets/3836616/362136f3-7407-4a92-8cb9-2aaeb8b77393)

**Comment by sumitzarkar** [comment]
@skrishn 

- To render the popup titles in feature-list we are using the popupUtils from the solutions-component. And, this is a known limitation for now.
- This needs to be fixed in the popupUtils in the solutions-component. 
- This utils function needs updates on different cases like date, domain, etc. Currently, it just returns the values of the attributes so it will work correctly only for string, and number fields.

@jmhauck - please let us know your thoughts on this, we think this should be handled in the same way as it is handled in the downloadUtils

**Comment by jmhauck** [comment]
@sumitzarkar yes...please do. Mike has done more to build things out in downloadUtils.
